### PR TITLE
feat: prototype chat interface

### PR DESCRIPTION
**Pull request by Hellebore**
## Checklist

- [ ] If `package.json` or `yarn.lock` have changed, then test the VSIX built by `yarn run vsce package` works from a direct install

### CODE REVIEW

File: media/search.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+//@ts-check
+
+// This script will be run within the webview itself
```
Comment (Hellebore) [inline media/search.js:3]: this file has just been renamed and formatted - it is the js behind the rule search functionality

File: media/vscode.css [hunk 2]
```diff
@@ -1,91 +1,91 @@
 :root {
-	--container-paddding: 20px;
-	--input-padding-vertical: 6px;
-	--input-padding-horizontal: 4px;
```
Comment (Hellebore) [inline media/vscode.css:4]: just reformatted

File: package.json [hunk 3]
```diff
@@ -51,6 +51,14 @@
 		},
 		"views": {
 			"sourcery-explorer": [
+				{
+					"id": "sourcery.chat",
+					"name": "Chat",
+					"type": "webview",
+					"icon": "sourcery-icon.png",
+					"contextualTitle": "Chat",
+					"when": "sourcery.features.code_understanding"
```
Comment (Hellebore) [inline package.json:60]: add the new view, only show it when the feature flag is set

**Review by brendanator — APPROVED** [review]
Looks good 👍

File: media/chat.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+//@ts-check
+
+// This script will be run within the webview itself
+// It cannot access the main VS Code APIs directly.
+(function () {
```
[the remark this answers is not in the document]
Comment (Hellebore) [reply media/chat.js:5]: Not generated, so I guess it could be yes

[the remark this answers is not in the document]
Comment (Hellebore) [reply media/chat.js]: Will do that separately

**Pull request merged**